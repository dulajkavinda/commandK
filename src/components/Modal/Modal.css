@keyframes fadein {
  from {
    transform: scale(0.8);
    opacity: 0;
  }
  to {
    transform: scale(1);
    opacity: 1;
  }
}

@keyframes bg-color {
  from {
    background-color: transparent;
  }
  to {
    background-color: rgba(#1e262f, 0.6);
  }
}

.modal-overlay {
  z-index: 99999;
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  background: rgba(#1e262f, 0.7);
  display: flex;
  justify-content: center;
  align-items: center;
  animation: bg-color 0.1s;
}

.modal-box {
  border: 1px solid #494e54;

  padding-left: 0.5em;
  padding-right: 0.5em;
  font-family: sans-serif;
}

.modal-box-medium {
  display: block;
  background: #1e262f;
  width: 43em;
  min-height: 30%;
  border-radius: 1rem;
  animation: fadein 0.2s;
}

.modal-box-small {
  display: block;
  background: #1e262f;
  width: 30em;
  min-height: 30%;
  padding: 1rem;
  border-radius: 1rem;
  animation: fadein 0.2s;
}

.modal-box-large {
  display: block;
  background: #1e262f;
  width: 70%;
  min-height: 30%;
  padding: 1rem;
  border-radius: 1rem;
  animation: fadein 0.2s;
}

.modal-box-header {
  border-top-left-radius: 1rem;
  border-top-right-radius: 1rem;
  border-bottom: 1px solid #494e54;
  display: flex;
  height: 3em;
  width: 100%;
}

.modal-box-header-search {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 100%;
}

.modal-box-header-search-left {
  display: flex;
  flex-direction: row;
}

.modal-box-header-search-results {
  color: #8f8e8e;
  margin-right: 5px;
  font-size: 1rem;
}

.modal-box-header-search-icon {
  display: flex;
  align-items: center;
  margin-left: 5px;
  margin-top: 1px;
}

.modal-box-header-search-username {
  color: #8095a9;
  font-size: 0.9rem;
  font-weight: 500;
  font-family: sans-serif;
  margin-left: 10px;
  margin-right: 10px;
  font-weight: 600;
  padding: 0.2em 0.8em;
  border-radius: 3px;

  background-color: #363e47;
}

.modal-box-header-search-input {
  border: none;
  outline: none;
  background: none;
  color: #d3d3d3;
  font-size: 1rem;
}

.modal-box-body {
  display: flex;
  flex-direction: column;

  overflow: scroll;
  overflow-y: auto;
  overflow-x: hidden;
  height: 20em;
}

.modal-box-body-item {
  width: 100%;
}

.modal-box-body-item-title {
  color: #8095a9;
  font-size: 0.9rem;
  font-weight: 500;
  font-weight: 600;
  padding-top: 1em;
  padding-bottom: 0.2em;
  margin-bottom: 10px;
  margin-left: 10px;
}

.modal-box-body-items:focus {
  outline: none;
  border: none;
  border: 1px dashed #687585;
  display: flex;
  flex-direction: row;
  align-items: center;
  color: #d3d3d3;
  background-color: #18222c;
  opacity: 0.7;
  -webkit-transition-property: opacity;
  -webkit-transition-duration: 0.5s;
  -webkit-transition-timing-function: cubic-bezier(0.15, 0.44, 0.47, 0.99);
  transition-property: opacity;
  transition-duration: 0.5s;
  transition-timing-function: cubic-bezier(0.15, 0.44, 0.47, 0.99);
}

.modal-box-body-items {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  color: #d3d3d3;
  padding: 0.8em 1em;
  border-radius: 10px;
  border: 1px solid transparent;
  cursor: pointer;
  background-color: #19222c;
  margin-top: 10px;
  margin-bottom: 10px;
  margin-right: 10px;
}

.modal-box-body-items:hover {
  opacity: 0.7;
  border: 1px dashed #687585;

  -webkit-transition-property: opacity;
  -webkit-transition-duration: 0.5s;
  -webkit-transition-timing-function: cubic-bezier(0.15, 0.44, 0.47, 0.99);
  transition-property: opacity;
  transition-duration: 0.5s;
  transition-timing-function: cubic-bezier(0.15, 0.44, 0.47, 0.99);
}

.modal-box-body-items-title {
  font-size: 1.1rem;
  font-weight: 600;
}

.modal-box-body-items-icon {
  margin-right: 10px;
}

.modal-box-body-items-left {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.modal-box-body-items-goto {
  display: flex;
  align-items: center;
  font-size: 0.9em;
  font-weight: 600;
  color: #687585;
}

.modal-box-body-items-goto-icon {
  margin-right: 5px;
  display: flex;
  align-items: center;
}

.modal-box-body-items-goto-text-start {
  margin-right: 5px;
}

.modal-box-body-items:not(:focus) .modal-box-body-items-goto {
  display: none;
}

.modal-box-header-search-right {
  display: flex;
  flex-direction: row;
}

.modal-box-header-search-right-nav {
  display: flex;
  flex-direction: row;

  justify-content: space-between;
  margin-right: 20px;
}

.modal-box-header-search-right-icon {
  margin-right: 7px;
}

.modal-box-header-search-right-text {
  font-size: 1em;
  font-weight: 600;
  color: #687585;
}

.modal-box-header-search-right-text-start {
  font-size: 1em;
  font-weight: 600;
  color: #687585;
  margin-right: 7px;
}

.modal-box-body::-webkit-scrollbar-thumb {
  background-color: #5d646b;
  border-radius: 20px;
}

.modal-box-body::-webkit-scrollbar {
  width: 8px;
}

.modal-box-body::-webkit-scrollbar-track {
  background: #1e262f;
}

/* mobile media queries */

@media (max-width: 768px) {
  .modal-overlay {
    align-items: end;
  }

  .modal-box-medium {
    width: 100%;
  }

  .modal-box-small {
    width: 100%;
  }

  .modal-box-large {
    width: 100%;
  }

  .modal-box-header-search-right-nav {
    display: none;
  }

  .modal-box-body-items-goto {
    display: none;
  }
}
