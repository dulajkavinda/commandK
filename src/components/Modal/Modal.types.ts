import { ReactNode } from 'react'

export type ModalSize = 'small' | 'medium' | 'large'

export interface ModalType {
  children?: ReactNode
  size?: ModalSize
  username?: string
  isOpen: boolean
  data: Group[]
  toggle: () => void
}
export interface State {
  selectedIndex: number
}

export type Action = { type: 'arrowUp' } | { type: 'arrowDown' } | { type: 'select'; payload: number }

export interface Item {
  icon: string
  title: string
  url?: string
}

export interface Group {
  sectionName: string
  items: Item[]
}
