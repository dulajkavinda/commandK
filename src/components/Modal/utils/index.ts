import SearchIcon from './SearchIcon'
import EnterKeyIcon from './EnterKeyIcon'
import DownArrowIcon from './DownArrowIcon'

export { SearchIcon, EnterKeyIcon, DownArrowIcon }
