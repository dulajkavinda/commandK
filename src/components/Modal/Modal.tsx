import React, { useEffect, useReducer, useRef, useMemo, useState } from 'react'
import classNames from 'classnames'
import { SearchIcon } from './SearchIcon'
import useKeyPress from '../../hooks/useKeyPress'
import { Action, Group, Item, ModalSize, ModalType, State } from './Modal.types'
import './Modal.css'

const Modal = (props: ModalType) => {
  const modalClass = classNames('modal-box', {
    [`modal-box-${props.size}`]: props.size,
  })

  const [list, setList] = useState<Group[] | []>(props.data || [])

  const inputRef = useRef<HTMLInputElement>(null)

  const arrowUpPressed = useKeyPress('ArrowUp')
  const arrowDownPressed = useKeyPress('ArrowDown')

  const initialState = { selectedIndex: 0 }

  const items = useMemo(() => {
    return list && list.map((item) => item.items).flat()
  }, [list])

  const filterItems = (list: Group[], title: string): void => {
    const result: Group[] = []

    list.forEach((obj) => {
      const items = obj.items.filter((item) => item.title.toLowerCase().includes(title.toLowerCase()))

      if (items.length > 0) {
        result.push({
          ...obj,
          items,
        })
      }
    })

    setList(result)
  }

  const reducer = (state: State, action: Action): State => {
    switch (action.type) {
      case 'arrowUp': {
        return {
          selectedIndex: state.selectedIndex,
        }
      }
      case 'arrowDown': {
        const button = document.getElementsByClassName(`modal-box-body-items-${state.selectedIndex}`)[0] as HTMLElement

        if (button) {
          button.focus()
        }

        return {
          selectedIndex: state.selectedIndex !== items.length - 1 ? state.selectedIndex + 1 : 0,
        }
      }
      case 'select':
        return { selectedIndex: action.payload }
      default:
        throw new Error()
    }
  }

  const [, dispatch] = useReducer(reducer, initialState)

  useEffect(() => {
    if (inputRef.current) {
      const x = window.scrollX
      const y = window.scrollY
      inputRef.current.focus({
        preventScroll: true,
      })
      window.scrollTo(x, y)
    }
  }, [])

  useEffect(() => {
    if (arrowUpPressed) {
      dispatch({ type: 'arrowUp' })
    }
  }, [arrowUpPressed])

  useEffect(() => {
    if (arrowDownPressed) {
      dispatch({ type: 'arrowDown' })
    }
  }, [arrowDownPressed])

  let trackItemindex = -1

  return (
    <>
      {props.isOpen && (
        <div className='modal-overlay' onClick={props.toggle}>
          <div onClick={(e) => e.stopPropagation()} id='modal-box' className={modalClass}>
            <div className='modal-box-header'>
              <div className='modal-box-header-search'>
                <div className='modal-box-header-search-left'>
                  <div className='modal-box-header-search-icon'>
                    <SearchIcon />
                    <span className='modal-box-header-search-username'>{props.username}</span>
                  </div>
                  <input
                    onChange={(e) => {
                      filterItems(props.data, e.target.value)
                    }}
                    ref={inputRef}
                    className='modal-box-header-search-input'
                    type='text'
                    placeholder='Search or jump to'
                    onKeyDown={(e) => {
                      if (e.key === 'ArrowDown') {
                        console.log('arrow down')
                      }
                    }}
                  />
                </div>

                <div className='modal-box-header-search-results'>{items.length} Results</div>
              </div>
            </div>
            <div className='modal-box-body'>
              {list &&
                list.map((item: Group, index) => {
                  return (
                    <div key={index} className='modal-box-body-item'>
                      <div className='modal-box-body-item-title'>{item.sectionName}</div>
                      {item.items.map((subItem: Item) => {
                        const listItemIndex = trackItemindex++
                        return (
                          <div
                            role='button'
                            tabIndex={0}
                            key={trackItemindex}
                            onKeyDown={(e) => {
                              if (e.key === 'Enter') {
                                window.location.href = `${subItem.url}`
                              }
                            }}
                            onClick={() => {
                              window.location.href = `${subItem.url}`
                            }}
                            className={classNames('modal-box-body-items', {
                              [`modal-box-body-items-${listItemIndex + 1}`]: true,
                            })}
                          >
                            <div className='modal-box-body-items-left'>
                              <div className='modal-box-body-items-icon'>{subItem.icon}</div>
                              <div className='modal-box-body-items-title'>{subItem.title}</div>
                            </div>
                            <span className='modal-box-body-items-goto'>Go to</span>
                          </div>
                        )
                      })}
                    </div>
                  )
                })}
            </div>
          </div>
        </div>
      )}
    </>
  )
}

Modal.defaultProps = {
  size: 'medium' as ModalSize,
  data: [],
}

export default Modal
