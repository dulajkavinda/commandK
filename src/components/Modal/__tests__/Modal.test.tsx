import React from 'react'
import { screen, render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Modal from '../Modal'
import { Group } from '../Modal.types'

const list: Group[] = [
  {
    sectionName: 'Repositories',
    items: [
      {
        icon: '🗂',
        title: 'How to make a good repo random',
        url: '/repo',
      },
      {
        icon: '🚀',
        title: 'This is another one to check',
        url: 'https://google.lk',
      },
    ],
  },
  {
    sectionName: 'Projects',
    items: [
      {
        icon: '🏋️‍♀️',
        title: 'How to make a good repo',
      },
      {
        icon: '📸',
        title: 'This is another one to check',
      },
    ],
  },
  {
    sectionName: 'Something',
    items: [
      {
        icon: '🏋️‍♀️',
        title: 'How to make a good repo',
        url: 'https://google.lk',
      },
      {
        icon: '📸',
        title: 'This is another one to check',
        url: 'https://google.lk',
      },
      {
        icon: '🏋️‍♀️',
        title: 'How to make a good repo',
        url: 'https://google.lk',
      },
      {
        icon: '📸',
        title: 'This is another one to check',
        url: 'https://google.lk',
      },
    ],
  },
]

describe('components/Modal', () => {
  it('renders modal', () => {
    const modal = renderer.create(<Modal data={list} isOpen toggle={() => null} />).toJSON()
    expect(modal).toMatchSnapshot()
  })

  it('renders modal with given data', () => {
    render(<Modal data={list} isOpen toggle={() => null} />)
    const modal = screen.getByTestId('modal')

    expect(modal).toHaveTextContent('Repositories')
    expect(modal).toHaveTextContent('Projects')
    expect(modal).toHaveTextContent('Something')

    expect(modal).toHaveTextContent('How to make a good repo')
    expect(modal).toHaveTextContent('This is another one to check')

    expect(modal).toHaveTextContent('🗂')
    expect(modal).toHaveTextContent('🚀')
    expect(modal).toHaveTextContent('🏋️‍♀️')
    expect(modal).toHaveTextContent('📸')
  })

  it('by default button should be rendered with medium size', () => {
    render(<Modal data={list} isOpen toggle={() => null} />)
    const modal = screen.getByTestId('modal')

    expect(modal).toHaveClass('modal-box-medium')
  })

  it('modal should be rendered with a given size', () => {
    render(<Modal size='small' data={list} isOpen toggle={() => null} />)
    const modal = screen.getByTestId('modal')

    expect(modal).toHaveClass('modal-box-small')
  })

  it('by default button should be rendered with username gome', () => {
    render(<Modal data={list} isOpen toggle={() => null} />)
    const modal = screen.getByTestId('modal')

    expect(modal).toHaveTextContent('home')
  })

  it('modal should be rendered with a given username', () => {
    render(<Modal username='adam' data={list} isOpen toggle={() => null} />)
    const modal = screen.getByTestId('modal')

    expect(modal).toHaveTextContent('adam')
  })

  it('modal should be closed when overlay is clicked', () => {
    const onClick = jest.fn()
    render(<Modal username='adam' data={list} isOpen toggle={onClick} />)
    const modalOverlay = screen.getByTestId('modal-overlay')

    modalOverlay.click()
    expect(onClick).toHaveBeenCalled()
  })

  it('show Press text in modal if it is not small', () => {
    render(<Modal size='medium' data={list} isOpen toggle={() => null} />)
    const modal = screen.getByTestId('press-text')

    expect(modal).toHaveTextContent('Press')
  })

  it('hide Press text in modal if it is small', () => {
    render(<Modal size='small' data={list} isOpen toggle={() => null} />)

    expect(screen.queryByTestId('press-text')).not.toBeInTheDocument()
  })

  it('focus on the 1st item when the down arrow pressed', () => {
    render(<Modal size='small' data={list} isOpen toggle={() => null} />)

    const input = screen.getByTestId('search-input')

    input.focus()
    fireEvent.keyDown(input, { key: 'ArrowDown', code: 'ArrowDown' })

    expect(screen.getByTestId('modal')).toHaveTextContent('How to make a good repo')
  })

  it('filter list based on search terms', () => {
    render(<Modal size='small' data={list} isOpen toggle={() => null} />)

    const input = screen.getByTestId('search-input')

    input.focus()
    fireEvent.change(input, { target: { value: 'random' } })
    input.blur()

    expect(screen.getByTestId('modal')).toHaveTextContent('Repositories')
    expect(screen.queryByText('Projects')).not.toBeInTheDocument()
    expect(screen.queryByText('Something')).not.toBeInTheDocument()
  })
})
