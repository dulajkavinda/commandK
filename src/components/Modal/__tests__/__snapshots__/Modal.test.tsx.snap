// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Modal renders modal 1`] = `
<div
  className="modal-overlay"
  data-testid="modal-overlay"
  onClick={[Function]}
>
  <div
    className="modal-box modal-box-medium"
    data-testid="modal"
    id="modal-box"
    onClick={[Function]}
  >
    <div
      className="modal-box-header"
    >
      <div
        className="modal-box-header-search"
      >
        <div
          className="modal-box-header-search-left"
        >
          <div
            className="modal-box-header-search-icon"
          >
            <svg
              fill="none"
              height={18}
              viewBox="0 0 21 21"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 20L15.514 15.506M18 9.5C18 11.7543 17.1045 13.9163 15.5104 15.5104C13.9163 17.1045 11.7543 18 9.5 18C7.24566 18 5.08365 17.1045 3.48959 15.5104C1.89553 13.9163 1 11.7543 1 9.5C1 7.24566 1.89553 5.08365 3.48959 3.48959C5.08365 1.89553 7.24566 1 9.5 1C11.7543 1 13.9163 1.89553 15.5104 3.48959C17.1045 5.08365 18 7.24566 18 9.5Z"
                stroke="#707680"
                strokeLinecap="round"
                strokeWidth="2"
              />
            </svg>
            <span
              className="modal-box-header-search-username"
            >
              home
            </span>
          </div>
          <input
            className="modal-box-header-search-input"
            data-testid="search-input"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search or jump to"
            type="text"
          />
        </div>
        <div
          className="modal-box-header-search-right"
        >
          <div
            className="modal-box-header-search-right-nav"
          >
            <span
              className="modal-box-header-search-right-text-start"
              data-testid="press-text"
            >
              Press
            </span>
            <div
              className="modal-box-header-search-right-icon"
            >
              <svg
                data-testid="down-arrow-icon-18"
                height={18}
                viewBox="0 0 14 14"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  transform="rotate(180 7 7)"
                >
                  <g
                    fill="none"
                    stroke="#707680"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  >
                    <rect
                      height="13"
                      rx="1"
                      width="13"
                      x=".5"
                      y=".5"
                    />
                    <path
                      d="M8.5 10.5v-3h2L7 3.5l-3.5 4h2v3h3z"
                    />
                  </g>
                </g>
              </svg>
            </div>
            <span
              className="modal-box-header-search-right-text"
            >
              to Navigate
            </span>
          </div>
          <div
            className="modal-box-header-search-results"
            data-testid="noresults"
          >
            8 Results
          </div>
        </div>
      </div>
    </div>
    <div
      className="modal-box-body"
    >
      <div
        className="modal-box-body-item"
      >
        <div
          className="modal-box-body-item-title"
        >
          Repositories
        </div>
        <div
          className="modal-box-body-items modal-box-body-items-0"
          data-testid="modal-box-body-item"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="modal-box-body-items-left"
          >
            <div
              className="modal-box-body-items-icon"
              data-testid="modal-box-body-items-icon"
            >
              üóÇ
            </div>
            <div
              className="modal-box-body-items-title"
            >
              How to make a good repo random
            </div>
          </div>
          <div
            className="modal-box-body-items-goto"
          >
            <span
              className="modal-box-body-items-goto-text-start"
            >
              Press
            </span>
            <div
              className="modal-box-body-items-goto-icon"
            >
              <svg
                fill="none"
                height={18}
                viewBox="0 0 15 15"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 1H2C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1Z"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M6 11L4 9L6 7"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M4 9H9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8V5"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <span
              className="modal-box-body-items-goto-text"
            >
              to Jump
            </span>
          </div>
        </div>
        <div
          className="modal-box-body-items modal-box-body-items-1"
          data-testid="modal-box-body-item"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="modal-box-body-items-left"
          >
            <div
              className="modal-box-body-items-icon"
              data-testid="modal-box-body-items-icon"
            >
              üöÄ
            </div>
            <div
              className="modal-box-body-items-title"
            >
              This is another one to check
            </div>
          </div>
          <div
            className="modal-box-body-items-goto"
          >
            <span
              className="modal-box-body-items-goto-text-start"
            >
              Press
            </span>
            <div
              className="modal-box-body-items-goto-icon"
            >
              <svg
                fill="none"
                height={18}
                viewBox="0 0 15 15"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 1H2C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1Z"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M6 11L4 9L6 7"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M4 9H9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8V5"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <span
              className="modal-box-body-items-goto-text"
            >
              to Jump
            </span>
          </div>
        </div>
      </div>
      <div
        className="modal-box-body-item"
      >
        <div
          className="modal-box-body-item-title"
        >
          Projects
        </div>
        <div
          className="modal-box-body-items modal-box-body-items-2"
          data-testid="modal-box-body-item"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="modal-box-body-items-left"
          >
            <div
              className="modal-box-body-items-icon"
              data-testid="modal-box-body-items-icon"
            >
              üèãÔ∏è‚Äç‚ôÄÔ∏è
            </div>
            <div
              className="modal-box-body-items-title"
            >
              How to make a good repo
            </div>
          </div>
          <div
            className="modal-box-body-items-goto"
          >
            <span
              className="modal-box-body-items-goto-text-start"
            >
              Press
            </span>
            <div
              className="modal-box-body-items-goto-icon"
            >
              <svg
                fill="none"
                height={18}
                viewBox="0 0 15 15"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 1H2C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1Z"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M6 11L4 9L6 7"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M4 9H9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8V5"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <span
              className="modal-box-body-items-goto-text"
            >
              to Jump
            </span>
          </div>
        </div>
        <div
          className="modal-box-body-items modal-box-body-items-3"
          data-testid="modal-box-body-item"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="modal-box-body-items-left"
          >
            <div
              className="modal-box-body-items-icon"
              data-testid="modal-box-body-items-icon"
            >
              üì∏
            </div>
            <div
              className="modal-box-body-items-title"
            >
              This is another one to check
            </div>
          </div>
          <div
            className="modal-box-body-items-goto"
          >
            <span
              className="modal-box-body-items-goto-text-start"
            >
              Press
            </span>
            <div
              className="modal-box-body-items-goto-icon"
            >
              <svg
                fill="none"
                height={18}
                viewBox="0 0 15 15"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 1H2C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1Z"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M6 11L4 9L6 7"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M4 9H9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8V5"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <span
              className="modal-box-body-items-goto-text"
            >
              to Jump
            </span>
          </div>
        </div>
      </div>
      <div
        className="modal-box-body-item"
      >
        <div
          className="modal-box-body-item-title"
        >
          Something
        </div>
        <div
          className="modal-box-body-items modal-box-body-items-4"
          data-testid="modal-box-body-item"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="modal-box-body-items-left"
          >
            <div
              className="modal-box-body-items-icon"
              data-testid="modal-box-body-items-icon"
            >
              üèãÔ∏è‚Äç‚ôÄÔ∏è
            </div>
            <div
              className="modal-box-body-items-title"
            >
              How to make a good repo
            </div>
          </div>
          <div
            className="modal-box-body-items-goto"
          >
            <span
              className="modal-box-body-items-goto-text-start"
            >
              Press
            </span>
            <div
              className="modal-box-body-items-goto-icon"
            >
              <svg
                fill="none"
                height={18}
                viewBox="0 0 15 15"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 1H2C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1Z"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M6 11L4 9L6 7"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M4 9H9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8V5"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <span
              className="modal-box-body-items-goto-text"
            >
              to Jump
            </span>
          </div>
        </div>
        <div
          className="modal-box-body-items modal-box-body-items-5"
          data-testid="modal-box-body-item"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="modal-box-body-items-left"
          >
            <div
              className="modal-box-body-items-icon"
              data-testid="modal-box-body-items-icon"
            >
              üì∏
            </div>
            <div
              className="modal-box-body-items-title"
            >
              This is another one to check
            </div>
          </div>
          <div
            className="modal-box-body-items-goto"
          >
            <span
              className="modal-box-body-items-goto-text-start"
            >
              Press
            </span>
            <div
              className="modal-box-body-items-goto-icon"
            >
              <svg
                fill="none"
                height={18}
                viewBox="0 0 15 15"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 1H2C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1Z"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M6 11L4 9L6 7"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M4 9H9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8V5"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <span
              className="modal-box-body-items-goto-text"
            >
              to Jump
            </span>
          </div>
        </div>
        <div
          className="modal-box-body-items modal-box-body-items-6"
          data-testid="modal-box-body-item"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="modal-box-body-items-left"
          >
            <div
              className="modal-box-body-items-icon"
              data-testid="modal-box-body-items-icon"
            >
              üèãÔ∏è‚Äç‚ôÄÔ∏è
            </div>
            <div
              className="modal-box-body-items-title"
            >
              How to make a good repo
            </div>
          </div>
          <div
            className="modal-box-body-items-goto"
          >
            <span
              className="modal-box-body-items-goto-text-start"
            >
              Press
            </span>
            <div
              className="modal-box-body-items-goto-icon"
            >
              <svg
                fill="none"
                height={18}
                viewBox="0 0 15 15"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 1H2C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1Z"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M6 11L4 9L6 7"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M4 9H9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8V5"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <span
              className="modal-box-body-items-goto-text"
            >
              to Jump
            </span>
          </div>
        </div>
        <div
          className="modal-box-body-items modal-box-body-items-7"
          data-testid="modal-box-body-item"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="modal-box-body-items-left"
          >
            <div
              className="modal-box-body-items-icon"
              data-testid="modal-box-body-items-icon"
            >
              üì∏
            </div>
            <div
              className="modal-box-body-items-title"
            >
              This is another one to check
            </div>
          </div>
          <div
            className="modal-box-body-items-goto"
          >
            <span
              className="modal-box-body-items-goto-text-start"
            >
              Press
            </span>
            <div
              className="modal-box-body-items-goto-icon"
            >
              <svg
                fill="none"
                height={18}
                viewBox="0 0 15 15"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 1H2C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1Z"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M6 11L4 9L6 7"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <path
                  d="M4 9H9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8V5"
                  stroke="#707680"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <span
              className="modal-box-body-items-goto-text"
            >
              to Jump
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
